libcloud/storage/drivers/s3.py

```diff
@@ -942,14 +942,28 @@ class S3USWestOregonStorageDriver(S3StorageDriver):
     ex_location_name = 'us-west-2'
 
 
-class S3CNNorthConnection(S3Connection):
+class S3CNNorthConnection(SignedAWSConnection, BaseS3Connection):
     host = S3_CN_NORTH_HOST
+    service_name = 's3'
+    version = API_VERSION
+
+    def __init__(self, user_id, key, secure=True, host=None, port=None,
+                 url=None, timeout=None, proxy_url=None, token=None,
+                 retry_delay=None, backoff=None):
+        super(S3CNNorthConnection, self).__init__(
+            user_id, key, secure, host,
```

Comment:
Minor thing - please use keyword arguments when calling a method (`user_id=user_id, key=key`, etc.). It makes code less error prone (now it's easy to accidentally break it if the argument order changes or similar).